src/discord/commands/book.ts

```diff
@@ -0,0 +1,229 @@
+import { CommandInteraction, MessageEmbed, MessageActionRow, MessageSelectMenu, MessageButton, ButtonInteraction, SelectMenuInteraction, Message } from 'discord.js';
+import roomModel, { Room } from '../../models/room.model';
+import sectionModel, { Section } from '../../models/section.model';
+import Timeblock from '../../models/timeblock.model';
+import { DateTime } from 'luxon';
+import { Types } from 'mongoose';
+
+interface TimeblockInformation {
+    startsAt: Date;
+    endsAt: Date;
+    availableCapacity: number;
+}
+
+interface sectionInformation extends Section {
+    _id?: Types.ObjectId;
+}
+
+function dateSuffix(day: number) {
+    if (day > 3 && day < 21) return 'th';
+    switch (day % 10) {
+        case 1:
+            return 'st';
+        case 2:
+            return 'nd';
+        case 3:
+            return 'rd';
+        default:
+            return 'th';
+    }
+}
+
+function getDateOptions(selectedDate?: string) {
+    const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+    const dateOptions = [];
+    const date = new Date();
+
+    for (let i = 0; i < 14; i++) {
+        dateOptions.push({
+            label: `${days[date.getDay()]} - ${months[date.getMonth()]} ${date.getDate()}${dateSuffix(date.getDate())}`,
+            value: date.toLocaleDateString().split('/').reverse().join('-'),
+            default: date.toLocaleDateString().split('/').reverse().join('-') === selectedDate ? true : false,
+        });
+        date.setDate(date.getDate() + 1);
+    }
+    return dateOptions;
+}
+
+async function parseCommandOptions(interaction: CommandInteraction): Promise<string[] | undefined> {
+    //Parses the book command option parameters to return the corresponding Room's section ID
+    const roomName = interaction.options.getString('room-name');
+    const roomJson = await roomModel.findOne({ name: roomName !== null ? roomName : undefined });
+
+    if (roomJson === null) {
+        interaction.reply({ content: 'Invalid Room: Do `/rooms` to see all avaliable rooms!', ephemeral: true });
+        return undefined;
+    }
+
+    const sectionJson = await sectionModel.find({ roomId: roomJson!._id });
+    const specificSection = sectionJson.find((s) => s.name === interaction.options.getString('section-name'));
+
+    if (specificSection === undefined) {
+        interaction.reply({ content: 'Invalid Section: Do `/rooms` to see all avaliable rooms and sections!', ephemeral: true });
+        return undefined;
+    }
+
+    return [roomJson._id, specificSection?._id];
```

Comment:
I think you can use `specificSection._id` since you check if `specificSection` is undefined already.